scripts/enable_fips.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+# https://aws.amazon.com/blogs/publicsector/enabling-fips-mode-amazon-linux-2/
+if [[ "$ENABLE_FIPS_MODE" == "true" ]]; then
+  # install and enable fips modules
+  sudo yum install -y dracut-fips openssl
+  sudo dracut -f
+
+  # enable fips in the boot command
+  sudo /sbin/grubby --update-kernel=ALL --args="fips=1"
+  sudo sed -i 's/^\(GRUB_CMDLINE_LINUX_DEFAULT=.*\)"$/\1 fips=1"/' /etc/default/grub
+
+  # rebuild grub
+  sudo grub2-mkconfig -o /etc/grub2.cfg
```

Comment:
I think the `grubby` command is the only required bit:
```suggestion
  sudo /sbin/grubby --update-kernel=ALL --args="fips=1"
```

Reply:
When I launch an instance with the latest 1.27 AMI, I see the other flags as expected:
```
cat /proc/cmdline 
BOOT_IMAGE=/boot/vmlinuz-5.10.192-183.736.amzn2.x86_64 root=UUID=11ca5c14-da9a-4407-a25a-475704058dc1 ro console=tty0 console=ttyS0,115200n8 net.ifnames=0 biosdevname=0 nvme_core.io_timeout=4294967295 rd.emergency=poweroff rd.shell=0 LANG=en_US.UTF-8 KEYTABLE=us psi=1 clocksource=tsc tsc=reliable
```

Which base AMI are you using?

Reply:
My env is 1.24 at the moment.
```
2023-10-09T09:46:25-06:00:     amazon-ebs: Adding tag: "build_region": "us-east-2"
2023-10-09T09:46:25-06:00:     amazon-ebs: Adding tag: "docker_version": "20.10.23-1.amzn2.0.1"
2023-10-09T09:46:25-06:00:     amazon-ebs: Adding tag: "kubernetes": "1.24.17/2023-09-14/bin/linux/x86_64"
2023-10-09T09:46:25-06:00:     amazon-ebs: Adding tag: "source_ami_id": "ami-0e6220111a9f780d6"
2023-10-09T09:46:25-06:00:     amazon-ebs: Adding tag: "source_ami_name": "amzn2-ami-minimal-hvm-2.0.20230926.0-x86_64-ebs"
2023-10-09T09:46:25-06:00:     amazon-ebs: Adding tag: "ssm_agent_version": "latest"
2023-10-09T09:46:25-06:00:     amazon-ebs: Adding tag: "created": "1696865624"
2023-10-09T09:46:25-06:00:     amazon-ebs: Adding tag: "containerd_version": "1.6.*"
2023-10-09T09:46:25-06:00:     amazon-ebs: Adding tag: "cni_plugin_version": "v1.2.0"
2023-10-09T09:46:25-06:00:     amazon-ebs: Adding tag: "Name": "amazon-eks-fips-node-1.24-v20231009"
```

Reply:
Oh it's related to the `sudo grub2-mkconfig -o /etc/grub2.cfg`. Let me try switching to just grubby as I took this from the other change.

Reply:
Ok so need to use grubby or grub-mkconfig but not both.